### PR TITLE
feat : Lesson12

### PR DESCRIPTION
**Pull request by AyakaOnozuka131**
# [Issue No.12](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#12)

**課題内容**
クリックしたらリクエストをして、それらが表示されるようにしてください

## [CodeSandBox](https://codesandbox.io/s/lesson12-00ns3?file=/index.js)
 :Latest commit [d4987c0](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/12/commits/d4987c0ab6c2551da3747e587dd9c66dafd66fef)

## 実装上、不安なところ、重点的に見てもらいたいところ、聞きたいこと

### CODE REVIEW

**Review by sae-github — APPROVED** [review]

**Comment by sae-github** [comment]
I thought it was very good, so I will approve it :)

File: lesson12/index.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+const ul = document.getElementById('js-list');
+const button = document.getElementById('js-button');
+
+const createLoading = () => {
+    const li = document.createElement("li");
+    const loadingImg = document.createElement('img');
+
+    li.style.listStyle = "none";
```
Comment (kenmori) [inline lesson12/index.js]: nits
use reset.css

File: lesson12/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+};
+
+/**
+ * Async関数
+ * fetchData関数から値が返ってきた後、その値を返す
+ */
+const fetchListData = async () => {
+    createLoading();
+    try {
+        const data = await fetchData();
```
Comment (kenmori) [inline lesson12/index.js]: recommend
```suggestion
        const data = await fetchData(jsonUrl);
```

File: lesson12/index.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    ul.appendChild(li);
+    console.error(error.message);
+};
+
+/**
+ * Async関数 
+ * @return {Array} 
+ */
+const jsonUrl = 'https://myjson.dit.upm.es/api/bins/5tqv';
+const fetchData = async () => {
```
Comment (kenmori) [inline lesson12/index.js]:
```suggestion
const fetchData = async (endpoint) => {
```

File: lesson12/index.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    try {
+        const response = await fetch(jsonUrl) // 中身はresponseオブジェクトを含むpromise
```
Comment (kenmori) [inline lesson12/index.js]:
```suggestion
        const response = await fetch(endpoint) // 中身はresponseオブジェクトを含むpromise
```

**Review by kenmori — APPROVED** [review]

**Comment by kenmori** [comment]
Approve

File: lesson12/index.js (hunk 1 above)
Comment (AyakaOnozuka131) [reply lesson12/index.js]: Moved to CSS
[ab8332a](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/12/commits/ab8332a31196cfc93765e86b503f281fce2b2e70)

File: lesson12/index.js (hunk 2 above)
Comment (AyakaOnozuka131) [reply lesson12/index.js]: Thank you for review!
Changed the json value passing part.
[d4987c0](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/12/commits/d4987c0ab6c2551da3747e587dd9c66dafd66fef)

**Review by kenmori — APPROVED** [review]

**Pull request merged**